repositories.md

```diff
@@ -5,10 +5,74 @@ css_id: repositories
 
+## Secure Source of Time
+
+Without access to a secure source of time, ECUs may be prevented from receiving the most recent updates. If the ECU's time is too high, the ECU will detect that the current valid metadata is expired and will be unable to perform an update. If the ECU's time is too low, an attacker would be able to replay old metadata to the ECU and install an old, potentially insecure, version.
+
+To prevent these issues, ECUs need access to a secure source of time. If an ECU does not have a secure clock, we recommend the use of a Time Server for time attestations. This section describes how a Time Server can be used in an Uptane implementation.
+
+### Time server
+
+A Time Server is a server that is responsible for the distribution of a secure source of time.
+
+The Time Server exists to inform ECUs about the current time in a cryptographically secure way, since many ECUs in a vehicle do not have a reliable source of time. The Time Server receives a list of tokens from vehicles, and returns back a list of signed records containing every token in the original list of tokens received and at least one instance of the current time.
+
+If the Time Server is used, it MUST conform to the following requirements:
+
+* When the Time Server receives a sequence of tokens from a vehicle, it SHALL provide one or more signed responses, containing the time along with these tokens. It MAY produce either one signed time attestation containing the current time and all tokens, or multiple time attestations each containing the current time and one or more tokens.
+
+* The Time Server SHALL expose a public interface allowing primaries to communicate with it. This communication MAY occur over FTP, FTPS, SFTP, HTTP, or HTTPS.
+
+* Rotation of the The Time Server's key is performed by listing the new key in the Director's Root metadata, in the same manner as other role keys are listed, and also in the Director's Targets metadata (for partial verification secondaries).
+
+#### Changes to the Director repository
+If a Time Server is in use, a representation of the Time Server public key is CONDITIONALLY REQUIRED in Director repository root metadata.
+
+If a Time Server is implemented AND partial-verification secondaries are used, the following metadata is CONDITIONALLY REQUIRED in the Director repository's Targets metadata:
+
+* A representation of the public key(s) for the Time Server, similar to the representation of public keys in Root metadata.
+
+Listing the public key of the Time Server in Director targets metadata is necessary to allow partial-verification secondaries to perform time server key rotation.
+
+#### Changes to a Primary
+
+If the Time Server is implemented, the primary SHALL use the following procedure to verify the time. This procedure occurs after the vehicle version manifest is sent and will fulfill the "Download and check current time" step of the Uptane Standard.
```

Comment:
I'm going to go through probably today to fix the MUSTs in all the pages, per our discussion in last week's meeting. However, since Marina has been shepherding this PR, I think I should let her fix the SHALLS and MUSTs in this file to avoid any possible conflict.